sdks/java/core/src/main/java/org/apache/beam/sdk/testing/WindowFnTestUtils.java

```diff
@@ -197,9 +229,21 @@ public void merge(Collection<W> otherWindows, W window) {
    */
   public static <T, W extends BoundedWindow> void validateNonInterferingOutputTimes(
       WindowFn<T, W> windowFn, long timestamp) throws Exception {
-    Collection<W> windows = WindowFnTestUtils.<T, W>assignedWindows(windowFn, timestamp);
+    // lift Timestamp into TimestampedValue to factorize implementation
+    validateNonInterferingOutputTimesWithValue(windowFn,
+        TimestampedValue.of((T) null, new Instant(timestamp)));
+  }
+  /**
+   * Assigns the given {@code timestampedValue} to windows using the specified {@code windowFn}, and
+   * verifies that result of {@code windowFn.getOutputTimestamp} for each window is within the
+   * proper bound. This version allows passing a {@link TimestampedValue}
+   * in case the value is needed to assign windows.
+   */
+  public static <T, W extends BoundedWindow> void validateNonInterferingOutputTimesWithValue(
+      WindowFn<T, W> windowFn, TimestampedValue<T> timestampedValue) throws Exception {
+    Collection<W> windows = assignedWindowsWithValue(windowFn, timestampedValue);
```

Comment:
In this method the value is used in `assignedWindowsWithValue()` which creates a `TestAssignContext` initialized with a proper `timestampedValue` that has non-null value set. It is there to allow assigning elements to windows based on values as well (see `WindowTest#testMergingCustomWindows`). If I move the code to a method that only takes the timestamp, I'm unable to do assign windows based on values anymore.